src/api/omnipool.ts

```diff
@@ -26,102 +23,30 @@ export type TOmnipoolAssetsData = Array<{
 }>
 
 export const useOmnipoolDataObserver = () => {
-  const { data, isLoading } = useQuery<TOmnipoolAssetsData>(
-    WS_QUERY_KEYS.omnipoolAssets,
+  const { data: omnipoolTokens, isLoading: isOmnipoolTokensLoading } =
+    useQuery<TOmnipoolAssetsData>(QUERY_KEYS.omnipoolTokens, {
+      enabled: false,
+      staleTime: Infinity,
+    })
+
+  const { data: hubToken, isLoading: isHubTokenLoading } = useQuery<PoolToken>(
+    QUERY_KEYS.hubToken,
     {
       enabled: false,
       staleTime: Infinity,
     },
   )
 
   return {
-    data,
-    dataMap: data ? new Map(data.map((asset) => [asset.id, asset])) : undefined,
-    isLoading,
+    data: omnipoolTokens,
+    hubToken,
+    dataMap: omnipoolTokens
+      ? new Map(omnipoolTokens.map((asset) => [asset.id, asset]))
```

Comment:
I would create the map inside `select` callback on useQuery to avoid creating new instances of the map on each render.

Reply:
but inside select prop the Map instance will be created on each render it's not cached. And I know how to get array of map values, just don't want to do in each component, but here